pkg/cli/helm/helm.go

```diff
@@ -136,3 +172,17 @@ func runUpgrade(upgradeClient *helm.Upgrade, releaseName string, helmChart *char
 	}
 	return err
 }
+
+// extractHelmError is a helper function to extract a specific error
+// (403 unauthorized when downloading a helm chart from ghcr.io) from a chain of errors.
+// If the error is not found, it returns nil.
+func extractHelmError(err error) error {
+	var errUnexpectedStatus containerderrors.ErrUnexpectedStatus
+	if errors.As(err, &errUnexpectedStatus) {
+		if errUnexpectedStatus.StatusCode == http.StatusForbidden && strings.Contains(errUnexpectedStatus.RequestURL, "ghcr.io") {
+			return fmt.Errorf("recieved 403 unauthorized when downloading helm chart from the registry. you may want to perform a `docker logout ghcr.io` and re-try the command")
+		}
+	}
+
+	return nil
```

Comment:
We need to also handle the case where the error is NOT an `containerderrors.ErrUnexpectedStatus`. Think about a case where the user is not connected to the internet. 

Every time we special case one type of error, we need to also handle the case where it's some other type of error.

Reply:
Oh, nevermind. I see you did that above.

Reply:
I'd suggest putting this line in here as well for clarity: https://github.com/radius-project/radius/pull/7455/files#diff-35b8f26ef00e63853d6a8303dfdb8755355d7cb6d48ba9449d5a04af62a723cbR142